File: posinterface/src/main/java/com/interswitch/posinterface/posshim/PrintConfiguration.java
package com.interswitch.posinterface.posshim;

import android.graphics.Bitmap;

import java.util.ArrayList;
import java.util.List;

public class PrintConfiguration {

    private List<Object> printObjects = new ArrayList<>();

    class Line {}

    public PrintConfiguration addString(String data) {
        printObjects.add(data);
        return this;
    }

    public PrintConfiguration addBitmap(Bitmap bitmap) {
        printObjects.add(bitmap);
        return this;
    }

    public PrintConfiguration addLine() {
        printObjects.add(new Line());
        return this;
    }

    public List<Object> getPrintObjects() {
        return printObjects;
    }
}

File: interswitchpossdk/src/main/java/com/interswitchng/interswitchpossdk/shared/models/posconfig/PrintConfiguration.kt
package com.interswitchng.interswitchpossdk.shared.models.posconfig

import android.graphics.Bitmap


internal class PrintConfiguration {

    private val printObjects = mutableListOf<PrintObject>()

    fun addData(data: String): PrintConfiguration {
        printObjects.add(PrintObject.Data(data))
        return this
    }

    fun addBitmap(bitmap: Bitmap): PrintConfiguration {
        printObjects.add(PrintObject.BitMap(bitmap))
        return this
    }

    fun addLine(): PrintConfiguration {
        printObjects.add(PrintObject.Line())
        return this
    }

    fun getPrintObjects(): List<Any> {
        return printObjects
    }
}